jmx/src/main/java/io/airlift/jmx/JmxAgent9.java

```diff
@@ -43,27 +44,71 @@
     public JmxAgent9(JmxConfig config)
             throws IOException
     {
+        Integer existingRegistryPort = Integer.getInteger(JMX_REGISTRY_PORT);
+        Integer existingServerPort = Integer.getInteger(JMX_SERVER_PORT);
+
+        if ((existingRegistryPort != null) && (config.getRmiRegistryPort() != null) &&
+                (!existingRegistryPort.equals(config.getRmiRegistryPort()))) {
+            throw new RuntimeException(format("System property [%s] value [%s] does match configured RMI registry port [%s]",
+                    JMX_REGISTRY_PORT, existingRegistryPort, config.getRmiRegistryPort()));
+        }
+
+        if ((existingServerPort != null) && (config.getRmiServerPort() != null) &&
+                (!existingServerPort.equals(config.getRmiServerPort()))) {
+            throw new RuntimeException(format("System property [%s] value [%s] does match configured RMI server port [%s]",
+                    JMX_SERVER_PORT, existingServerPort, config.getRmiServerPort()));
+        }
+
         int registryPort;
-        if (config.getRmiRegistryPort() == null) {
```

Comment:
It's a little hard to follow the flow of this if block now. I think it might be more readable to spell out the conditions and write it as follows:

```java
int registryPort;
if (existingRegistryPort != null && config.getRmiRegistryPort() == null) {
    registryPort = existingRegistryPort;
}
else if (existingRegistryPort == null && config.getRmiRegistryPort() != null) {
    registryPort = config.getRmiRegistryPort();
}
else {
    registryPort = NetUtils.findUnusedPort();
}
```

(or some other form using firstNonNull)